windows-agent/internal/proservices/proservices_test.go

```diff
@@ -63,15 +64,25 @@ func TestNew(t *testing.T) {
 			}
 
 			s, err := proservices.New(ctx, publicDir, privateDir, proservices.WithRegistry(reg))
-			if err == nil {
-				defer s.Stop(ctx)
-			}
-
 			if tc.wantErr {
 				require.Error(t, err, "New should return an error")
 				return
 			}
 			require.NoError(t, err, "New should return no error")
+			defer s.Stop(ctx)
```

Comment:
The issue with your current change is that if we are in a test case where we wanted an error and got none, the service will be created and not stopped.

Reply:
Good point!

Reply:
You shouldn’t stop if the err is not nil (I didn’t recheck the code if it’s safe, but even if it is today, it might not be next time, so unfortunately, you need to check if err == nil which isn’t really idiomatic, I agree.